src/modules/servers/application/validators/conditional-priority.validator.ts

```
@@ -0,0 +1,71 @@
+      return 'Priority must be a number between 1 and 999 if provided';
+    }
+    return 'Priority is required and must be a number between 1 and 999';
+  }
+}
+
+/**
+ * Decorator that validates priority conditionally based on server type
+ * @param validationOptions Optional validation options
+ */
+export function IsConditionalPriority(validationOptions?: ValidationOptions) {
+  return function (object: Object, propertyName: string) {
```

Biome (1.9.4)

[error] Don't use 'Object' as a type.

Prefer explicitly define the object shape. This type means "any non-nullable value", which is slightly better than 'unknown', but it's still a broad type.

(lint/complexity/noBannedTypes)